pkg/argo/fetch_test.go

```diff
@@ -277,26 +272,28 @@ func TestGetPodLogs(t *testing.T) {
 		t.Skip("argo: skip workflow tests")
 	}
 	a := assert.New(t)
-	_, err := kubectlDeleteFromYAML(podYAML)
-	a.NoError(err)
 	podID, err := kubectlCreateFromYAML(podYAML)
 	a.NoError(err)
+	defer kubectlDeletePod(podID)
 
 	err = waitUntilPodRunning(podID)
 	a.NoError(err)
 	offset := ""
-	realLogs := []string{}
+	actual := []string{}
+	expected := []string{"hello1", "hello2", "hello3"}
 	for {
+		isPodCompleted := isPodCompleted(podID)
```

Comment:
Get the latest pod status can fix the random failed.

Reply:
Good catch.